servers/zts/src/main/java/com/yahoo/athenz/zts/store/impl/S3ChangeLogStore.java

```diff
@@ -71,24 +76,30 @@ public boolean supportsFullRefresh() {
     
     @Override
     public SignedDomain getSignedDomain(String domainName) {
-        
-        // make sure we have an aws s3 client for our request
-        
-        if (awsS3Client == null) {
-            awsS3Client = getS3Client();
-        }
-        
-        SignedDomain signedDomain = getSignedDomain(awsS3Client, domainName);
-
-        // if we got a failure for any reason, we're going
-        // get a new aws s3 client and try again
-        
+        // clear the mapping if present and value is stored else null is returned
+        SignedDomain signedDomain = tempSignedDomainMap.remove(domainName);
+        // when for some reason the getAllSignedDomains() was unsuccessful signedDomain will be null
         if (signedDomain == null) {
-            awsS3Client = getS3Client();
+            // make sure we have an aws s3 client for our request
+
+            if (awsS3Client == null) {
+                awsS3Client = getS3Client();
+            }
+
             signedDomain = getSignedDomain(awsS3Client, domainName);
+
+            // if we got a failure for any reason, we're going
+            // get a new aws s3 client and try again
+
+            if (signedDomain == null) {
+                awsS3Client = getS3Client();
+                signedDomain = getSignedDomain(awsS3Client, domainName);
+            }
+
+            return signedDomain;
```

Comment:
let's replace lines 99-102 with just a single return, we should have:

if (signedDomain == null) { 
....
}

return signedDomain;